djangorestframework_camel_case/util.py

```diff
@@ -35,10 +36,13 @@ def camelize(data):
                 new_key = re.sub(camelize_re, underscore_to_camel, key)
             else:
                 new_key = key
-            new_dict[new_key] = camelize(value)
+            if key not in ignore_fields and new_key not in ignore_fields:
+                new_dict[new_key] = camelize(value, **options)
+            else:
+                new_dict[new_key] = value
```

Comment:
Shouldn't this be:

```py
new_dict[key] = value
```

instead?

Otherwise, the new key (post transformation) is applied, which is not what is expected.

For example, using `__typename` in `ignore_fields` won't work here.

Reply:
What about something like this instead?

```py
def camelize (data, **options):

    ignore_fields = options.get('ignore_fields') or ()

    if isinstance(data, Promise):
        data = force_text(data)

    if isinstance(data, dict):

        if isinstance(data, ReturnDict):
            new_dict = ReturnDict(serializer = data.serializer)
        else:
            new_dict = OrderedDict()

        for (key, value) in data.items():

            if isinstance(key, Promise):
                key = force_text(key)

            if key in ignore_fields:
                new_dict[key] = camelize(value, **options)

            else:

                if isinstance(key, str) and ('_' in key):
                    new_key = re.sub(camelize_re, underscore_to_camel, key)
                else:
                    new_key = key

                new_dict[new_key] = camelize(value, **options)

        return new_dict

    if is_iterable(data) and not isinstance(data, str):
        return [camelize(item, **options) for item in data]

    return data
```

Reply:
> Shouldn't this be:
> 
> ```python
> new_dict[key] = value
> ```
> 
> instead?
> 
> Otherwise, the new key (post transformation) is applied, which is not what is expected.
> 
> For example, using `__typename` in `ignore_fields` won't work here.

Thanks for your comment! It could be that the naming is a bit confusing. What we really are doing is ignoring fields under the given key. In other words, we don't want to recursively modify any of the data under the specified key. We might need to split the option of not changing specific key names (like `__typename` in your example) with not modifying any data under certain key names (like `extraData` or `extra_data` in my example, where the field name should be changed but the others shouldn't).

Reply:
I am okay with this being called `ignore_fields` if we want to support the other model maybe call it `ignore_keys`